restcomm/restcomm.testsuite/src/main/java/org/restcomm/connect/testsuite/http/util/CustomDnsResolver.java

```diff
@@ -0,0 +1,75 @@
+/*
+ *  TeleStax, Open Source Cloud Communications
+ *  Copyright 2011-2018, Telestax Inc and individual contributors
+ *  by the @authors tag.
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation; either version 3 of
+ *  the License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>
+ */
+
+package org.restcomm.connect.testsuite.http.util;
+
+
+import java.net.InetAddress;
+import java.net.UnknownHostException;
+import java.util.List;
+import java.util.Map;
+
+/**
+ * @author gvagenas@gmail.com
+ */
+
+public class CustomDnsResolver implements sun.net.spi.nameservice.NameService {
+    //Based on https://stackoverflow.com/a/43870031
+
+    private static String LOCALHOST = "127.0.0.1";
+    private static Map<String, String> domainIpMap;
```

Comment:
`domainIpMap` never seems to be initialised in the code.

Does it need to be static? what about converting this class into a singleton instead?

Reply:
@hrosa it is initialized in the org.restcomm.connect.testsuite.http.util.CustomDnsResolver#setNameService and needs to be static because that method is static